CMakeLists.txt

```diff
@@ -298,6 +298,8 @@ if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
   # For O_CLOEXEC, O_DIRECTORY, and O_NOFOLLOW flags on older systems
   # (pre POSIX.1-2008: glibc 2.11 and earlier). #4042
   add_definitions(-D_GNU_SOURCE)
+elseif(NOT WIN32)
+    add_definitions(-D_XOPEN_SOURCE=700)
```

Comment:
too much indent, also would be good to have a comment:

    # For ptsname(). #6743

Reply:
Let's add it for now. But will it be updated when something else is added, that depends on XOPEN?

Reply:
Maybe or not, but it's good to have these low-cost hints. If they grow to list many things, then the hints aren't needed anymore, but until then it's worth it.

Reply:
But the point is it might not grow, because no one will ever notice the flag was necessary for the next thing... And `glibc` docs seem to lazy with documenting flags needed for a function, thus not noticing that way either.